### PR TITLE
Fix release workflow

### PR DESCRIPTION
**Pull request by venkatBala**
- [ ] I have added the tests to cover my changes.
- [ ] I have updated the documentation and CHANGELOG accordingly.
- [ ] I have read the CONTRIBUTING document.

### CODE REVIEW

**Review by wjcunningham7 — APPROVED** [review]

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -308,26 +308,31 @@ jobs:
...
-        run: |
-          docker build -t covalent .
-          docker tag covalent:latest public.ecr.aws/covalent/covalent:latest
-          docker tag covalent:latest public.ecr.aws/covalent/covalent:$TAG
-          if [ "${{ github.events.inputs.test_release }}" == false ] ; then
-            docker push -a public.ecr.aws/covalent/covalent
-          fi
-      - name: Update stable tag
+          github.event_name == 'workflow_call'
+          && github.events.inputs.pre_release == 'true'
```
Comment (scottwn) [inline .github/workflows/release.yml:320]:
```suggestion
          && github.events.inputs.prerelease == 'true'
```

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -308,26 +308,31 @@ jobs:
...
+      - name: Login to Public Registry
+        uses: docker/login-action@v1
+        with:
+          registry: public.ecr.aws
+          username: ${{ secrets.AWS_KEY_ID }}
+          password: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+      - name: Build and push pre-release
         if: >
-          inputs.prerelease
-          && ${{ !github.events.inputs.stable_version }}
```
Comment (scottwn) [inline .github/workflows/release.yml:315]: I wouldn't remove this double check

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -308,26 +308,31 @@ jobs:
...
+          context: .
+          file: ./Dockerfile
+          platforms: all
+          push: true
+          tags: |
+            public.ecr.aws/covalent/covalent:latest
+            public.ecr.aws/covalent/covalent:${{ env.TAG }}
+      - name: Tag stable release
         if: >
-          github.event.inputs.stable_version
```
Comment (scottwn) [inline .github/workflows/release.yml:325]: we should keep this; we only want to do the stable release if the `stable_version` exists

**Review by scottwn — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -308,26 +308,34 @@ jobs:
...
-        run: |
-          docker build -t covalent .
-          docker tag covalent:latest public.ecr.aws/covalent/covalent:latest
-          docker tag covalent:latest public.ecr.aws/covalent/covalent:$TAG
-          if [ "${{ github.events.inputs.test_release }}" == false ] ; then
-            docker push -a public.ecr.aws/covalent/covalent
-          fi
-      - name: Update stable tag
+          && github.event_name == 'workflow_call'
+          && github.events.inputs.prerelease == 'true'
```
Comment (scottwn) [inline .github/workflows/release.yml:322]: actually maybe we don't need this at all? `inputs.prerelease` is already checked in line 320

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -308,26 +308,34 @@ jobs:
...
+            public.ecr.aws/covalent/covalent:${{ env.TAG }}
+      - name: Tag stable release
         if: >
-          github.event.inputs.stable_version
-          && !inputs.prerelease
-          && steps.docker-build-push.outcome == 'success'
-          && ${{ !github.event.inputs.test_release }}
+          github.event_name == 'workflow_dispatch'
+          && github.event.inputs.stable_version
+          && !github.events.inputs.test_release
```
Comment (scottwn) [inline .github/workflows/release.yml:337]: what's the reason for this syntax change?

**Review by scottwn — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yml (hunk 5 above)
Comment (wjcunningham7) [reply .github/workflows/release.yml:337]: the ${{}} are not needed in `if` clauses

File: .github/workflows/release.yml [hunk 6]
```diff
@@ -308,26 +308,33 @@ jobs:
...
+          file: ./Dockerfile
+          platforms: all
+          push: true
+          tags: |
+            public.ecr.aws/covalent/covalent:latest
+            public.ecr.aws/covalent/covalent:${{ env.TAG }}
+      - name: Tag stable release
         if: >
-          github.event.inputs.stable_version
-          && !inputs.prerelease
```
Comment (scottwn) [inline .github/workflows/release.yml:326]: do we not want to also keep this check?

**Review by scottwn — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yml (hunk 6 above)
Comment (wjcunningham7) [reply .github/workflows/release.yml:326]: @venkatBala please add that line back in

Comment (venkatBala) [reply .github/workflows/release.yml:302]: added

**Comment by venkatBala** [comment]
@scottwn this should be good to go thanks!

**Review by scottwn — APPROVED** [review]

**Pull request merged**